pkg/graveler/graveler.go

```diff
@@ -1711,7 +1711,10 @@ func (g *Graveler) Set(ctx context.Context, repository *RepositoryRecord, branch
 
 		// verify the key not found
 		_, err := g.Get(ctx, repository, Ref(branchID), key)
-		if err == nil || !errors.Is(err, ErrNotFound) {
+		if err == nil { // Entry found, return precondition failed
```

Comment:
This changes behaviour of regular putObject, doesn't it?

Reply:
This was a bug

Reply:
Please confirm with one of the authors of the original feature -- we've historically had this feature with this behaviour and the opposite successful behaviour.  So I worry someone might depend on it.

https://hyrumslaw.com FTL.